server/core/src/rodsServer.cpp

```diff
@@ -305,6 +312,129 @@ namespace
 
         return 0;
     } // create_pid_file
+
+    void create_stacktrace_directory()
+    {
+        namespace fs = boost::filesystem;
+        boost::system::error_code ec;
+        fs::create_directories("../.." / fs::path{IRODS_DEFAULT_PATH_HOMEDIR} / "stacktrace", ec);
+    } // create_stacktrace_directory
+
+    void log_stacktrace_files(std::atomic<bool>& _stop_token)
+    {
+        namespace fs = boost::filesystem;
+
+        // The leading "../.." enables support for non-package installations.
+        const auto stacktrace_directory = "../.." / fs::path{IRODS_DEFAULT_PATH_HOMEDIR} / "stacktrace";
+
+        // Get the sleep time.
+        const auto sleep_time = [] {
+            try {
+                const auto adv_settings = irods::server_properties::instance().map()
+                    .at(irods::CFG_ADVANCED_SETTINGS_KW);
+
+                const auto iter = adv_settings
+                    .find(irods::CFG_STACKTRACE_FILE_PROCESSOR_SLEEP_TIME_IN_SECONDS_KW);
+
+                if (iter != std::end(adv_settings)) {
+                    if (const auto seconds = iter->get<int>(); seconds > 0) {
+                        return seconds;
+                    }
+                }
+            }
+            catch (...) {}
+
+            return 3; // The default sleep time.
```

Comment:
is 3 good?  perhaps 10?  or 30?   let's leave a justfication in a comment if we have one.

Reply:
No reason. It can be whatever we want. The larger it is, the later it will appear in the log.

Reply:
right, but balanced against how often this code fires and walks the directory for files to process.

3 feels quick to me.   @kript you have a good feel for this?  what's a default value that you wouldn't feel the immediate need to change?

Reply:
do we have a preferred default?

Reply:
10?

that's 3x slower than '3' (so less churn), but not so slow that it's WAY far away from *when* it happened.

Reply:
going with 10 seconds